monitoring/prober/scd/test_operation_simple_heavy_traffic_concurrent.py

```diff
@@ -0,0 +1,365 @@
+@default_scope(SCOPE_SC)
+def test_get_ops_by_search_concurrent(scd_session):
+  op_resp_map = {}
+  total_found_ids = set()
+
+  # Query opetions concurrently
+  with ThreadPoolExecutor(max_workers=THREAD_COUNT) as executor:
+    for idx in range(len(OP_IDS)):
+      future = executor.submit(_query_operation, idx, scd_session)
+      future.add_done_callback(functools.partial(_collect_resp_callback, idx, op_resp_map))
+
+  for idx, resp in op_resp_map.items():
+    assert resp.status_code == 200, resp.content
+    found_ids = [op['id'] for op in resp.json().get('operation_references', [])]
+    total_found_ids.update(found_ids)
+
+  assert len(_intersection(OP_IDS, total_found_ids)) == len(OP_IDS)
+
+
+# Preconditions: Operations with ids in OP_IDS created by scd_session user
+# Mutations: Operations with ids in OP_IDS mutated to second version
+@default_scope(SCOPE_SC)
+def test_mutate_ops_concurrent(scd_session):
+  op_req_map = {}
+  op_resp_map = {}
+  op_map = {}
+
+  # Build mutate requests
+  for idx, op_id in enumerate(OP_IDS):
+    op_req_map[op_id] = _build_mutate_request(idx, op_id, op_map, scd_session)
+  assert len(op_req_map) == len(OP_IDS)
+
+  # Mutate operations in parallel
+  with ThreadPoolExecutor(max_workers=THREAD_COUNT) as executor:
+    for op_id in OP_IDS:
+      req = op_req_map[op_id]
+      future = executor.submit(_put_operation, req, op_id, scd_session)
+      future.add_done_callback(functools.partial(_collect_resp_callback, op_id, op_resp_map))
+
+  ovn_map.clear()
+
+  for op_id, resp in op_resp_map.items():
+    existing_op = op_map[op_id]
+    assert existing_op
+
+    assert resp.status_code == 200, resp.content
+    data = resp.json()
+    op = data['operation_reference']
+    assert op['id'] == op_id
+    assert op['uss_base_url'] == 'https://example.com/uss2'
+    assert op['version'] == 2
+    assert op['subscription_id'] == existing_op['subscription_id']
+    assert 'state' not in op
+
+    ovn_map[op_id] = op['ovn']
+
+  assert len(ovn_map) == len(OP_IDS)
+
+
+# Preconditions: Operations with ids in OP_IDS mutated to second version
+# Mutations: Operations with ids in OP_IDS deleted
+def test_delete_op_concurrent(scd_session):
+  op_resp_map = {}
+
+  # Delete operations concurrently
+  with ThreadPoolExecutor(max_workers=THREAD_COUNT) as executor:
+    for op_id in OP_IDS:
+      future = executor.submit(_delete_operation, op_id, scd_session)
+      future.add_done_callback(functools.partial(_collect_resp_callback, op_id, op_resp_map))
+
+  assert len(op_resp_map) == len(OP_IDS)
+
+  for resp in op_resp_map.values():
+    assert resp.status_code == 200, resp.content
+
+
+# Preconditions: None
+# Mutations: None
+@default_scope(SCOPE_SC)
+def test_ops_do_not_exist_get(scd_session):
+  for op_id in OP_IDS:
+    resp = scd_session.get('/operation_references/{}'.format(op_id))
+    assert resp.status_code == 404, resp.content
+
+
+# Preconditions: None
+# Mutations: None
+@default_scope(SCOPE_SC)
```

Comment:
I think we can also remove this test -- the test above verifies that the operations no longer exist generally, and test coverage elsewhere ensures that requesting by get and by query return consistent results.

Reply:
Make sense, deleted.